app/decorators/lib/delayed/worker.rb

```diff
@@ -21,25 +26,56 @@ def perform(job)
 
   def set_task_protection(state)
     return unless ENV['ECS_AGENT_URI']
+    return unless state != @@task_protection_state
     begin
-      ecs_agent_uri = ENV['ECS_AGENT_URI']
-      state_endpoint = "#{ecs_agent_uri}/task-protection/v1/state"
-      uri = URI.parse(state_endpoint)
-      request = Net::HTTP::Put.new(uri)
-      request['Content-Type'] = 'application/json'
-      request.body = { 'ProtectionEnabled' => state }.to_json
-
-      response = Net::HTTP.start(uri.hostname, uri.port) do |http|
-        http.request(request)
+      if state
+        cancel_deactivation_task
+        send_task_protection_to_ecs(true) if state
+      else
+        debounced_task_protection_deactivation
       end
 
-      unless response.is_a?(Net::HTTPSuccess)
-        raise "Failed to update ECS task protection state: #{response.body}"
-      end
+      @@task_protection_state = state
 
     rescue StandardError => e
       raise "An error occurred: #{e.message}"
     end
+  end
+
+  def debounced_task_protection_deactivation
+    return if @@deactivation_pending
+    @@deactivation_pending = true
+    @@deactivation_task = Concurrent::ScheduledTask.execute(60) do
+      @@mutex.synchronize do
+        if @@active_jobs == 0
+          send_task_protection_to_ecs(false)
+          @@deactivation_pending = false
```

Comment:
I think this deactivation pending might need to be outside active_jobs == 0.

Reply:
Change made